src/strife/f_wipe.c

```diff
@@ -268,17 +281,25 @@ wipe_ScreenWipe
     {
 	go = 1;
         // haleyjd 20110629 [STRIFE]: We *must* use a temp buffer here.
-	wipe_scr = (byte *) Z_Malloc(width*height, PU_STATIC, 0); // DEBUG
+#ifndef CRISPY_TRUECOLOR
+	wipe_scr = (pixel_t *) Z_Malloc(width*height, PU_STATIC, 0); // DEBUG
 	//wipe_scr = I_VideoBuffer;
+#else
+	// [crispy] In TrueColor render perform everything in common buffer.
+	// Otherwise serious malloc errors will happen.
```

Comment:
Yes, I'm afraid. "Debug" build type can survive it, but "Release" type almost imideatelly crashes after crossfade or next hitting of `Z_Malloc` (like opening Load menu) with something like this:

<details>

```
Thread 1 received signal SIGSEGV, Segmentation fault.
0x00007ff6f69f396e in Z_Malloc ()
(gdb) bt
#0  0x00007ff6f69f396e in Z_Malloc ()
#1  0x00007ff6f6a07357 in M_StringAlloc.constprop.0 ()
#2  0x00007ff6f6a07c74 in M_SafeFilePath ()
#3  0x00007ff6f6a026f5 in M_ReadSaveStrings ()
#4  0x00007ff6f6a06415 in M_Responder ()
#5  0x00007ff6f69f698d in D_ProcessEvents ()
#6  0x00007ff6f69c6930 in BuildNewTic ()
#7  0x00007ff6f69c6afc in NetUpdate.part.0 ()
#8  0x00007ff6f69c7782 in TryRunTics ()
#9  0x00007ff6f69f6e9a in D_DoomLoop ()
#10 0x00007ff6f69f89c7 in D_DoomMain ()
#11 0x00007ff6f69c159f in SDL_main ()
#12 0x00007ff6f6a3f0fe in main_getcmdline ()
#13 0x00007ff6f69c12ee in __tmainCRTStartup () at C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:267
#14 0x00007ff6f69c13e6 in WinMainCRTStartup () at C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtexe.c:157
(gdb)

```

</details>

Since it's not a simple game of `PU_STATIC` vs. `PU_CACHE`, using of `I_VideoBuffer` is the best I can do.

Reply:
I'd like to have a closer look at this, probably this weekend, before I wave it through.

Reply:
This works for me:
```patch
--- a/src/strife/f_wipe.c
+++ b/src/strife/f_wipe.c
@@ -283,14 +283,8 @@ wipe_ScreenWipe
     {
        go = 1;
         // haleyjd 20110629 [STRIFE]: We *must* use a temp buffer here.
-#ifndef CRISPY_TRUECOLOR
-       wipe_scr = (pixel_t *) Z_Malloc(width*height, PU_STATIC, 0); // DEBUG
+       wipe_scr = (pixel_t *) Z_Malloc(width*height*sizeof(*wipe_scr), PU_STATIC, 0); // DEBUG
        //wipe_scr = I_VideoBuffer;
-#else
-       // [crispy] In TrueColor render perform everything in common buffer.
-       // Otherwise serious malloc errors will happen.
-       wipe_scr = I_VideoBuffer;
-#endif
        (*wipes[wipeno*3])(width, height, ticks);
     }
 
@@ -298,10 +292,8 @@ wipe_ScreenWipe
     V_MarkRect(0, 0, width, height);
     rc = (*wipes[wipeno*3+1])(width, height, ticks);
 
-#ifndef CRISPY_TRUECOLOR
     // haleyjd 20110629 [STRIFE]: Copy temp buffer to the real screen.
     V_DrawBlock(x, y, width, height, wipe_scr);
-#endif
 
     // final stuff
     if (rc)
```

Reply:
Ah, I have missed most important part just like in other parts of wiping functions - `*sizeof(*wipe_scr)`. Working on my side too, thank you! Will commit changes soon.

Reply:
Good point. Actually, `wipe_scr` is never freed in Doom as well, but we use `I_ScreenBuffer` unconditionally there.

Reply:
So memory leak is happening. 😧 Have a look at top-right "МБ" value from task manager, it stands for MB amount of used memory by executable. I'm just constantly loading a savegame to invoke crossfading:

https://github.com/fabiangreffrath/crispy-doom/assets/21193394/af32780a-823c-4874-9d28-b14813b49833

Guess, in wide screen modes and high resolution rendering consuption will be much higher, and once we hit reallocated ~512 MB zone memory limit, an "Z_Malloc: failed on reallocation of ..." will appear.

What is you recommendation how to proceed? Probably, two ways:
* Call `Z_Free(wipe_scr);` below [this line](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/strife/f_wipe.c#L233).
* Just use `I_VideoBuffer` like in Doom for both paletted and TrueColor renders for `wipe_scr`.

Reply:
I'd say let's go the Doom way.

Reply:
But for some reason there is this comment:
```C
// haleyjd 20110629 [STRIFE]: We *must* use a temp buffer here.
```

So, better explicitly call `Z_Free()`?

Reply:
> Call Z_Free(wipe_scr); below [this line](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/strife/f_wipe.c#L233).

Hm, didn't worked for me, that's odd. 🤔

Reply:
You right, turns out, it's not a wipe, it's something else. Even if wipe is disabled [here](https://github.com/fabiangreffrath/crispy-doom/blob/master/src/strife/d_main.c#L234) as:
```
    // save the current screen if about to wipe
    if (gamestate != wipegamestate && false) // +++ && false
```

...leak still happening. Probably something around with `malloc`?